modules/core/src/main/java/org/apache/synapse/mediators/eip/Target.java

```diff
@@ -270,4 +249,31 @@ public void setStatisticIdForMediators(ArtifactHolder holder){
         }
         StatisticIdentityGenerator.reportingEndBranchingEvent(holder);
     }
+
+    private boolean mediateMessage(SequenceMediator sequenceMediator, MessageContext synCtx) {
+        try {
+            return sequenceMediator.mediate(synCtx);
+        } catch (SynapseException syne) {
+            if (!synCtx.getFaultStack().isEmpty()) {
+                log.warn("Executing fault handler due to exception encountered");
```

Comment:
Isn't it better to add the exception stack traces?

Reply:
fixed with https://github.com/wso2/wso2-synapse/pull/1215